### PR TITLE
sqlbase,*: step 2 of removing sqlbase

### PR DESCRIPTION
**Pull request by ajwerner**
Part of #52519.

### CODE REVIEW

File: pkg/sql/catalog/descriptor.go [hunk 1]
```diff
@@ -11,10 +11,185 @@
...
+// SchemaDescriptor will eventually be called schemadesc.Descriptor.
+// It is implemented by ImmutableSchemaDescriptor.
+type SchemaDescriptor interface {
+	Descriptor
+	SchemaDesc() *descpb.SchemaDescriptor
+}
+
+// TableDescriptor is an interface around the table descriptor types.
+//
+// TODO(ajwerner): This interface likely belongs in a catalog/tabledesc package
```
Comment (rohany) [inline pkg/sql/catalog/descriptor.go:91]: can remove this todo now

**Review by rohany — APPROVED** [review]

**Review by thoszhang — APPROVED** [review]

**Review by ajwerner — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53081)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @lucy-zhang, and @rohany)

___
*[pkg/sql/catalog/descriptor.go, line 91 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/53081#-MF7cUdkAwNbxQMxuhV3-r5-91:-MFC820W06Dq9U3HEQW8:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/17de6cbd3d6cf3d12f2348c484930072fff30466/pkg/sql/catalog/descriptor.go#L91)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

can remove this todo now
</blockquote></details>

Done.
___
*[pkg/sql/sqlbase/table_desc.go, line 98 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/53081#-MF7cUddn51cF1tABUvT-r6-98:-MFC83MU9EX6XHsC6qqI:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/ea7678df6b55218499b511a282047919fe06f490/pkg/sql/sqlbase/table_desc.go#L98)):*
<details><summary><i>Previously, lucy-zhang (Lucy Zhang) wrote…</i></summary><blockquote>

I think this commit and the next one might be cruft from rebasing on top of #52497. They look like inverses of each other.
</blockquote></details>

Done.

**Comment by ajwerner** [comment]
TFTR!

bors r+

**Pull request merged**